File: examples/docs-code-snippets/src/main/java/examples/java/interactions/CheckStock.java
package examples.java.interactions;

import com.ing.baker.recipe.annotations.FiresEvent;
import com.ing.baker.recipe.annotations.RequiresIngredient;
import com.ing.baker.recipe.javadsl.Interaction;

import java.util.List;

public interface CheckStock extends Interaction {

    interface Outcome {
    }

    record OrderHasUnavailableItems(List<String> unavailableProductIds) implements Outcome {
    }

    record SufficientStock() implements Outcome {
    }

    @FiresEvent(oneOf = {SufficientStock.class, OrderHasUnavailableItems.class})
    Outcome apply(@RequiresIngredient("orderId") String orderId,
                  @RequiresIngredient("productIds") List<String> productIds
    );
}

File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/interactions/CheckStock.kt
package examples.kotlin.interactions

import com.ing.baker.recipe.javadsl.Interaction

interface CheckStock : Interaction {

    sealed interface Outcome

    data class OrderHasUnavailableItems(
        val unavailableProductIds: List<String>
    ) : Outcome

    object SufficientStock : Outcome

    fun apply(orderId: String, productIds: List<String>): Outcome
}
